microservices/case-service/src/main/java/org/entando/bundle/service/impl/FileServiceImpl.java

```diff
@@ -0,0 +1,239 @@
+package org.entando.bundle.service.impl;
+
+import org.apache.http.client.utils.URIBuilder;
+import org.entando.bundle.config.BundleConfiguration;
+import org.entando.bundle.service.FileService;
+import org.entando.bundle.service.impl.utils.FileHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.Delete;
+import software.amazon.awssdk.services.s3.model.DeleteObjectsRequest;
+import software.amazon.awssdk.services.s3.model.DeleteObjectsResponse;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.ListObjectsRequest;
+import software.amazon.awssdk.services.s3.model.ListObjectsResponse;
+import software.amazon.awssdk.services.s3.model.ObjectIdentifier;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectResponse;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+import software.amazon.awssdk.services.s3.model.S3Object;
+import software.amazon.awssdk.utils.StringUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+public class FileServiceImpl implements FileService {
+
+  private Logger log = LoggerFactory.getLogger(FileServiceImpl.class);
+
+  private final BundleConfiguration config;
+
+  public FileServiceImpl(BundleConfiguration config) {
+    this.config = config;
+  }
+
+  private String getBuketName() {
+    String name = null;
+
+    try {
+      name = config.getS3().getBucket().getName();
+    } catch (Throwable t) {
+      log.error("Configuration error detected! No bucket name defined", t.getLocalizedMessage());
+    }
+    return name;
+  }
+
+  private String getBuketUrl() {
+    String url = null;
+
+    try {
+      url = config.getS3().getBucket().getUrl();
+    } catch (Throwable t) {
+      log.error("Configuration error detected! No bucket url defined", t.getLocalizedMessage());
+    }
+    return url;
+  }
+
+  private S3Client getClient() {
+    // Create the S3Client object.
+    Region region = Region.EU_WEST_1;
```

Comment:
I think this should be taken from env var

Reply:
it slipped through, fixed